test/base/src/androidTest/kotlin/io/realm/kotlin/test/shared/RealmSetTests.kt

```diff
@@ -0,0 +1,1087 @@
+        }
+    }
+
+    @Test
+    fun contains() {
+        for (tester in managedTesters) {
+            tester.contains()
+        }
+    }
+
+    @Test
+    fun iterator() {
+        for (tester in managedTesters) {
+            tester.iterator()
+        }
+    }
+
+    @Test
+    fun iterator_hasNext() {
+        for (tester in managedTesters) {
+            tester.iterator_hasNext()
+        }
+    }
+
+    @Test
+    fun iterator_next() {
+        for (tester in managedTesters) {
+            tester.iterator_next()
+        }
+    }
+
+    @Test
+    fun iterator_remove() {
+        for (tester in managedTesters) {
+            tester.iterator_remove()
+        }
+    }
+
+    @Test
+    fun iterator_failsIfRealmClosed() {
+        // No need to be exhaustive
+        managedTesters[0].iteratorFailsIfRealmClosed(getCloseableRealm())
+    }
+
+    @Test
+    fun copyToRealm() {
+        for (tester in managedTesters) {
+            tester.copyToRealm()
+        }
+    }
+
+    @Test
+    fun add_detectsDuplicates() {
+        val leaf = Sample().apply { intField = 1 }
+        val child = Sample().apply {
+            intField = 2
+            nullableObject = leaf
+            objectSetField = realmSetOf(leaf, leaf)
+        }
+        realm.writeBlocking {
+            copyToRealm(Sample()).apply {
+                objectSetField.add(child)
+            }
+        }
+        val results = realm.query<Sample>().find()
+        assertEquals(3, results.size)
+    }
+
+    @Test
+    fun addAll_detectsDuplicates() {
+        val child = RealmSetContainer()
+        val parent = RealmSetContainer()
+        realm.writeBlocking {
+            copyToRealm(parent).apply {
+                objectSetField.addAll(setOf(child, child))
+            }
+        }
+        val results = realm.query<RealmSetContainer>().find()
+        assertEquals(2, results.size)
+    }
+
+    @Test
+    fun assign_updateExistingObjects() {
+        val parent = realm.writeBlocking {
+            copyToRealm(
+                SampleWithPrimaryKey().apply {
+                    primaryKey = 2
+                    objectSetField = realmSetOf(
+                        SampleWithPrimaryKey().apply {
+                            primaryKey = 1
+                            stringField = "INIT"
+                        }
+                    )
+                }
+            )
+        }
+        realm.query<SampleWithPrimaryKey>("primaryKey = 1")
+            .find()
+            .single()
+            .run {
+                assertEquals("INIT", stringField)
+            }
+
+        realm.writeBlocking {
+            findLatest(parent)!!.apply {
+                objectSetField = realmSetOf(
+                    SampleWithPrimaryKey().apply {
+                        primaryKey = 1
+                        stringField = "UPDATED"
+                    }
+                )
+            }
+        }
+        realm.query<SampleWithPrimaryKey>("primaryKey = 1")
```

Comment:
Shouldn't we also check that the `objectSetField` contains a Set with only 1 element with the UPDATED value?